src/lib/basic/FontParams.svelte

```diff
@@ -0,0 +1,137 @@
+<script context="module" lang="ts">
+  export type ChangeValue = { family: string | undefined; size: number | undefined };
+  export type ChangeEvent = { newValue: ChangeValue };
+</script>
+
+<script lang="ts">
+  import { createEventDispatcher, onDestroy, onMount } from "svelte";
+  import { tr } from "../../utils/i18n";
+  import MdIcon from "./MdIcon.svelte";
+  import { fontCache } from "../../stores";
+  import type { Unsubscriber } from "svelte/motion";
+
+  export let hasFamily: boolean = true;
+  export let hasSize: boolean = true;
+  export let family: string = "";
+  export let size: number = 0;
+  export let sizeMin: number = 1;
+  export let sizeMax: number = 999;
+  export let sizeStep: number = 2;
+
+  const dispatch = createEventDispatcher<{
+    change: ChangeEvent;
+  }>();
+
+  const fontQuerySupported = typeof queryLocalFonts !== "undefined";
+
+  let loading = false;
+  async function refreshFonts() {
+    if (fontQuerySupported) {
+      try {
+        loading = true;
+        fontCache.set(await queryLocalFonts());
+      } finally {
+        loading = false;
+      }
+    }
+  }
+
+  let families: Record<string, string> = {};
+  function resetFamilyAndStyle() {
+    if ($fontCache) {
+      families = {
+        "NotoSans-Regular": "Noto Sans - Regular",
+        ...$fontCache.reduce(
+          (pv, i) => {
+            pv[i.postscriptName] = `${i.family} - ${i.style}`;
+            return pv;
+          },
+          {} as typeof families,
+        ),
+      };
+    } else {
+      families = {
+        [family]: family,
+        "NotoSans-Regular": "Noto Sans - Regular",
+      };
+    }
+  }
+
+  function emitChange(newValue: Partial<ChangeValue>) {
+    dispatch("change", {
+      newValue: {
+        ...{ family: family, size },
+        ...newValue,
+      },
+    });
+  }
+
+  let unsubscriber: Unsubscriber;
+  onMount(() => {
+    resetFamilyAndStyle();
+    unsubscriber = fontCache.subscribe(resetFamilyAndStyle);
+  });
+  onDestroy(() => unsubscriber());
+</script>
+
+{#if hasFamily}
+  <div class="input-group flex-nowrap input-group-sm font-family">
+    <span class="input-group-text" title={$tr("params.text.font_family")}>
+      <MdIcon icon="text_format" />
+    </span>
+    {#if fontQuerySupported}
+      <select class="form-select" value={family} on:change={(e) => emitChange({ family: e.currentTarget.value })}>
+        {#each Object.keys(families).sort() as family}
+          <option value={family} style="font-family: {family}">{families[family]}</option>
```

Comment:
You are using wrong font names. Styles are not applied

![image](https://github.com/user-attachments/assets/52523b32-ed9c-418f-aafe-a914b7221a9a)